pkg/plugin/plugin_test.go

```diff
@@ -409,3 +409,53 @@ func TestValidateUpdateBlock(t *testing.T) {
 		t.Error("expected current cluster to pass update validation")
 	}
 }
+
+func TestGetPrivateAPIServerIPAddress(t *testing.T) {
```

Comment:
This test serves as decent documentation but we might also want to write down how this works somewhere for posterity's sake.

Reply:
i'll follow up with docs on the interface